docs/CodingGuidelines.md

```diff
@@ -0,0 +1,174 @@
+
+# Coding guidelines
+
+Guidelines that code authors and code reviewers are expected to adhere to in the development of IoT App Kit.
+
+## General guidelines
+
+1. Documentation must be written in a style that is consistent with the rest of the documentation (based on a AWS documentation style)  
+   Compare writing to other written examples within https://docs.aws.amazon.com/iot-sitewise/ as a primary source on how to write the documentation, as well as cross-referencing other documentation found within https://github.com/awslabs/iot-app-kit/tree/main/docs.
+
+1. Ensure that all documentation is correctly referenced in the table of contents, which can be found at https://github.com/awslabs/iot-app-kit/blob/main/README.md.
+
+   In-depth guidelines on documentation: https://alpha-docs-aws.amazon.com/awsstyleguide/latest/styleguide/awsstyleguide.pdf#aws-implementation-guides. This can be a useful reference, but you are not expected to read through and be 100% adherent to these guidelines.
+1. Components must have high quality testing, including integration tests which utilize the component in the way a user does. An example of a good example integration test can be found in the [@iot-app-kit/source-iotsitewise package](https://github.com/awslabs/iot-app-kit/blob/main/packages/source-iotsitewise/src/time-series-data/data-source.spec.ts#L562-L603)
+
+1. Tests should be written to maximize the chance that the test would actually catch a real issue, i.e. the likelihood of preventing a regression. This is the primary value of a test.  Useful talk on testing: https://www.youtube.com/watch?v=Fha2bVoC8SE3.
+
+2. Tests should also be written to minimize the pain they inflict on the development cycle. Every test represents some amount of burden, through the necessity to run it, maintain it, and alter the test as the codebase morphs, however some tests inflict more pain than others.  
+   *EXAMPLES OF PAIN*:
+  - tests failing when nothing broke
+  - flaky tests
+  - snapshot tests that vary on machine
+  - tests that require refactoring when irrelevant portions of the codebase are altered
+
+1. Tests should be written to not fail when things are not broken (this helps reduce inflicted pain). This can be achieved through techniques such as not over asserting.  
```

Comment:
Duplicate of the first "example of pain". Any way to combine these or change the formatting a bit?

Reply:
see https://github.com/awslabs/iot-app-kit/pull/208#discussion_r945917917